trace-utils/src/send_data/mod.rs

```diff
@@ -187,7 +192,7 @@ impl SendData {
     /// # Returns
     ///
     /// A `SendDataResult` instance containing the result of the operation.
-    pub async fn send(self) -> SendDataResult {
+    pub async fn send(&self) -> SendDataResult {
         if self.use_protobuf() {
```

Comment:
This part seemed a little bit odd because we're using the api key in the endpoint to distinguish the method to send the payload. Would it be better to match the collection type? WDYT?

Reply:
On each review I did, using the api key was stressing me. I needed to check twice that we cannot be in a case where we set the api key inadvertedly and use agentless. Overall, I'd rather have a clear parameter `isAgentless`. That said I don't think it's in the scope of that PR.

Reply:
+100